source/library/CabalGild/Type/SomeParsecParser.hs

```diff
@@ -9,22 +9,38 @@ module CabalGild.Type.SomeParsecParser where
 
 import qualified CabalGild.Type.List as List
 import qualified CabalGild.Type.Set as Set
+import qualified Distribution.CabalSpecVersion as CabalSpecVersion
 import qualified Distribution.Parsec as Parsec
 import qualified Distribution.Pretty as Pretty
+import qualified Text.PrettyPrint as PrettyPrint
 
-data SomeParsecParser
-  = forall c s b a.
-    (Parsec.Parsec (c s b a), Pretty.Pretty (c s b a)) =>
-    SomeParsecParser (Parsec.ParsecParser (c s b a))
+-- | This type bundles up a parser and a pretty printer for some type.
+-- Typically they will just be monomorphic versions of 'Parsec.parsec' and
+-- 'Pretty.prettyVersioned'.
+data SomeParsecParser = forall a.
+  SomeParsecParser
+  { parsec :: Parsec.ParsecParser a,
+    pretty :: CabalSpecVersion.CabalSpecVersion -> a -> PrettyPrint.Doc
+  }
```

Comment:
As I was writing the documentation for this type, I realized it could be dramatically simplified.